Sources/Datadog/DatadogInternal/Context/DeviceInfo.swift

```diff
@@ -69,13 +77,19 @@ extension DeviceInfo {
         processInfo: ProcessInfo = .processInfo,
         device: UIDevice = .current
     ) {
+        var architecture = "unknown"
+        if let archInfo = NXGetLocalArchInfo()?.pointee {
+            architecture = String(utf8String: archInfo.name) ?? "unknown"
+        }
```

Comment:
I wonder if these problems do still exist (these are dated back to 2020):
* https://github.com/getsentry/sentry-cocoa/issues/635
* https://github.com/firebase/firebase-ios-sdk/issues/6128

TL;DR; This was causing app rejection in App Store submission. Any thoughts @fuzzybinary @maxep ? Is there any alternative way to get this information?

Reply:
I actually saw those, but if I read things right, it looks like this only affected the Mac app store / Catalyst apps (which we don't support at the moment anyway?)

But even according to the Firebase issue, they didn't actually remove the call and haven't gotten any reports of being rejected since September.

Reply:
:/ this worries me a bit, but it looks like Firebase is still using this API without any report. So we can try with dogfooding and see if we got a rejection from the app store, WDYT? We would surely make sure the app team is ok with it!
My concern is that we aim to support multiplatform, so this would add debt for macos. We are not there yet and I prefer unblocking flutter and tackle that later 👍

Reply:
My guess is that API was flagged as private by accident.